app/shared/ui-settings/src/commonMain/kotlin/ui/settings/tabs/theme/ThemePreferences.kt

```diff
@@ -0,0 +1,291 @@
+//                state.update(themeSettings.copy(isAmoled = checked))
+//            },
+//            title = { Text("AMOLED") },
+//            description = { Text("在深色模式下使用纯黑背景") },
+//        )
+    }
+
+    val colorList =
+        ((4..10) + (1..3)).map { it * 35.0 }.map { Color(Hct.from(it, 40.0, 40.0).toInt()) }
+
+    Group(title = { Text("调色板") }) {
+        if (LocalPlatform.current.isMobile()) {
+            val colors = colorList.chunked(4)
+            val pageCount = colors.size
+
+            val pagerState = rememberPagerState(
+                initialPage = colors.indexOfFirst { chunk ->
+                    chunk.any { it.toArgb() == themeSettings.seedColor }
+                }.let { if (it == -1) 0 else it },
+            ) {
+                pageCount
+            }
+
+            HorizontalPager(
+                modifier = Modifier.fillMaxWidth().clearAndSetSemantics {},
+                state = pagerState,
+                contentPadding = PaddingValues(horizontal = 12.dp),
+            ) { page ->
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                    horizontalArrangement = Arrangement.Center,
+                ) {
+                    colors[page].forEach { color ->
+                        ColorButtons(
+                            color = color,
+                            themeSettings = themeSettings,
+                            state = state,
+                        )
+                    }
+                }
+            }
+
+            Row(
+                Modifier
+                    .fillMaxWidth()
+                    .align(Alignment.CenterHorizontally)
+                    .padding(vertical = 12.dp)
+                    .clearAndSetSemantics {},
+                horizontalArrangement = Arrangement.Center,
+            ) {
+                repeat(pageCount) { iteration ->
+                    val color = if (pagerState.currentPage == iteration)
+                        MaterialTheme.colorScheme.primary
+                    else
+                        MaterialTheme.colorScheme.outlineVariant
+                    Box(
+                        modifier = Modifier
+                            .padding(2.dp)
+                            .size(6.dp)
+                            .clip(CircleShape)
+                            .background(color),
+                    )
+                }
+            }
+        } else {
+            FlowRow(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.Center,
+                // maxItemsInEachRow = 4
+            ) {
+                colorList.forEach { color ->
+                    ColorButtons(
+                        color = color,
+                        themeSettings = themeSettings,
+                        state = state,
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun ColorButtons(
```

Comment:
```suggestion
fun ColorButton(
```

复数形式听起来会多个按钮. 这里应当用单数.